### PR TITLE
Miscible keywords

### PR DESCRIPTION
**Pull request by totto82**
Add table support for the following keywords
MSFN
PMISC
MISC
SGCWMIS
SORWMIS

Also checkRange() is added to SimpleTable to check whether the input span [0,1]

### CODE REVIEW

**Comment by joakim-hove** [comment]
Good - tests and all!

File: opm/parser/eclipse/CMakeLists.txt [hunk 1]
```diff
@@ -276,6 +276,11 @@ EclipseState/Tables/VFPProdTable.hpp
 EclipseState/Tables/VFPInjTable.hpp
 EclipseState/Tables/TableManager.hpp
 EclipseState/Tables/TableContainer.hpp
+EclipseState/Tables/SorwmisTable.hpp
```
Comment (joakim-hove) [inline opm/parser/eclipse/CMakeLists.txt]: The new header `SorwmisTable.hpp` is not part of the PR; so right now it does not compile.

**Comment by joakim-hove** [comment]
Actually none of the new headers are added to the PR.

File: opm/parser/eclipse/EclipseState/Tables/SimpleTable.cpp [hunk 2]
```diff
@@ -153,6 +153,16 @@ void SimpleTable::checkMonotonic(const std::string& columnName,
     }
 }
 
+void SimpleTable::checkRange(const std::string& columnName)
```
Comment (joakim-hove) [inline opm/parser/eclipse/EclipseState/Tables/SimpleTable.cpp]: If you indeed intend to throw I would prefer if the method were named e.g. `assertRange()` - it should be safe to call a method named `checkRange()` without catching fire.

Comment (joakim-hove) [inline opm/parser/eclipse/EclipseState/Tables/SimpleTable.cpp]: The range `[0,1]` is not *that* canonical; rename the method `assertUnitRange()` - or alternatively take limits as input arguments to the method.

File: opm/parser/eclipse/EclipseState/Tables/TableManager.cpp [hunk 3]
```diff
@@ -145,6 +146,21 @@ namespace Opm {
         addTables( "RVVD", m_eqldims->getNumEquilRegions());
 
         {
+            size_t numMiscibleTables = ParserKeywords::MISCIBLE::NTMISC::defaultValue;
+            if (deck.hasKeyword<ParserKeywords::MISCIBLE>()) {
+                auto keyword = deck.getKeyword<ParserKeywords::MISCIBLE>();
+                auto record = keyword->getRecord(0);
+                numMiscibleTables =  static_cast<size_t>(record->getItem<ParserKeywords::MISCIBLE::NTMISC>()->getInt(0));
+            }
```
Comment (joakim-hove) [inline opm/parser/eclipse/EclipseState/Tables/TableManager.cpp]: I'm glad you are using the `ParserKeywords` classes - I hope it feels reasonably natural!

File: opm/parser/eclipse/IntegrationTests/ParseMiscible.cpp [hunk 4]
```diff
@@ -0,0 +1,262 @@
...
+std::strcpy(tooFewTables,miscibleTightData);
+std::strcat(tooFewTables,sorwmisData);
+
+// missing miscible keyword
+BOOST_CHECK_THROW (parser->parseString(sorwmisData, ParseMode()), std::invalid_argument );
+
+//too many tables
+BOOST_CHECK_THROW( parser->parseString(tooFewTables, ParseMode()), std::invalid_argument);
+
+char data[10000];
```
Comment (joakim-hove) [inline opm/parser/eclipse/IntegrationTests/ParseMiscible.cpp]: Can't you just use `std::string`?

File: opm/parser/eclipse/CMakeLists.txt (hunk 1 above)
Comment (totto82) [inline opm/parser/eclipse/CMakeLists.txt]: Sorry. I didn't notice.

**Comment by joakim-hove** [comment]
Good - running full build test now; will merge when all is green.
